src/main/java/uk/gov/cslearning/record/validation/validators/LearnerNotInvitedValidator.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.cslearning.record.validation.validators;
+
+import org.springframework.stereotype.Component;
+import uk.gov.cslearning.record.dto.InviteDto;
+import uk.gov.cslearning.record.service.InviteService;
+import uk.gov.cslearning.record.validation.annotations.LearnerNotInvited;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.nio.file.Paths;
+
+@Component
+public class LearnerNotInvitedValidator implements ConstraintValidator<LearnerNotInvited, InviteDto> {
+
+    private final InviteService inviteService;
+
+    public LearnerNotInvitedValidator(InviteService inviteService) {
+        this.inviteService = inviteService;
+    }
+
+    public void initialise(LearnerNotInvited constraint) {}
+
+    public boolean isValid(InviteDto invite, ConstraintValidatorContext context) {
```

Comment:
As above this would look a little cleaner as...
```
    public boolean isValid(InviteDto invite, ConstraintValidatorContext context) {
        String eventUid = Paths.get(invite.getEvent().getPath()).getFileName().toString();

        return !inviteService.findByEventIdAndLearnerEmail(eventUid, invite.getLearnerEmail()).isPresent();
    }
```